exercises/07-normalizr/src/modules/entities/entities-selectors.ts

```diff
@@ -0,0 +1,47 @@
+import { createSelector } from 'reselect'
```

Comment:
I mean nothing against this appraoch. I am writing the selectors in the same way but it's reaaaally long and hard to read.

Maybe https://twitter.com/sebastienlorber/status/1293536507739942920 could help?

Reply:
This solution (1) looks better 

```ts
export const getUserSkills = createSelector(
  getEntitiesState,
  getSkills,
  ({ userSkills }, skills) =>
    mapValues(userSkills, (userSkill) => ({
      ...userSkill,
      skill: skills[userSkill.skill],
    }))
)
```
but it uses the functionality of the `lodash` library. 

@novellizator May I use `mapValues` from `lodash`? We already use `lodash` for `mergeWith`, but I get a feeling that we try to minimize the usage of libraries which are not the subject of this training.

As an alternative we could consider this approach (2):
```ts
export const getUserSkills = createSelector(
  getEntitiesState,
  getSkills,
  ({ userSkills }, skills) =>
    Object.fromEntries(
      Object.entries(userSkills).map(([id, userSkill]) => [
        id,
        {
          ...userSkill,
          skill: skills[userSkill.skill],
        },
      ])
    )
)
```
It's more readable than `reduce` for me

Reply:
As a Swift developer, I prefer the version with mapValues, as we have it in our standard library ;-)

I tend to agree with @tomkis in that it's easier to read and mapValues is a function people will be using on their projects anyway (until javascript finally adds it into std)

Reply:
I prefer the `mapValues` version as well. I think it's fine to use `lodash`. Almost every project has a library like this.

Reply:
Rewritten with `mapValues` in 874d670

Added a hint to tutorial in b1f80de